crates/cxx-qt-gen/test_inputs/inheritance.rs

```diff
@@ -0,0 +1,34 @@
+#[cxx_qt::bridge]
+mod inheritance {
+    extern "C++" {
+        include!("cxx-qt-lib/qmodelindex.h");
+        type QModelIndex = cxx_qt_lib::QModelIndex;
+        include!("cxx-qt-lib/qvariant.h");
+        type QVariant = cxx_qt_lib::QVariant;
+    }
+
+    #[cxx_qt::qobject(base = "QAbstractItemModel")]
+    #[derive(Default)]
+    pub struct MyObject {
+        data: Vec<i32>,
+    }
+
+    impl qobject::MyObject {
+        #[qinvokable(cxx_override)]
+        pub fn data(&self, _index: &QModelIndex, _role: i32) -> QVariant {
+            QVariant::default()
+        }
+
+        cxx_qt::inherit! {
```

Comment:
the `passthrough_and_naming` test did have a custom base class already to test that. So you could reuse that one to test the inherits too to avoid creating more tests for us to maintain :-) (maybe it should be called something different)

Reply:
Right, I was hesitant to use that, as I'm not at all testing `passthrough_and_naming`.
I would rather remove the `#[qinvokable(cxx_inherit)]`, etc. tests from `passthrough_and_naming`, as that test-case seems to become quite large by now.

Reply:
right passthrough_and_naming has ended up testing "everything else"

- namespaces
- file stem
- correctly ignoring/passthrough CXX blocks
- base class
- invokable and property names to camel case work etc
- two qobjects in the same mod don't collide

So maybe an inheritance one is OK and "base class" could be removed from passthrough and naming :sweat_smile:

Reply:
I agree, we can remove that in a separate PR I'd say.